caffe2/python/mkl/rewrite_graph.py

```diff
@@ -21,6 +21,21 @@ def last_producer(ops, blob):
     raise ValueError("Failed to find last producer of blob, %s", blob)
 
 
+def fix_BoxWithNMSLimit(net):
+    outputs = set()
+    for op in net.op:
+        if op.type == 'BoxWithNMSLimit':
```

Comment:
This part is really hard coded to work around the 0-dim tensor issue in MKL-DNN:
https://github.com/pytorch/pytorch/blob/edb88b5f3af03718b443d015f195faa1832ce95b/caffe2/ideep/operators/operator_fallback_ideep.cc#L101-L103

Ideally we want all the output to be IDEEP tensor so that they stay in the same context during computation. We only avoid that if for those outputs that can be 0 dim, in this case input 0-2 but not 3.